src/Resources/UserSessionResource.php

```diff
@@ -201,6 +201,60 @@ protected function handleLogin(array $credentials = [], $remember = false)
         }
     }
 
+    /**
+     * @param array $credentials
+     * @return string
+    */
+    private function generateToken($credentials) {
+        $integrateio_id = $credentials['integrateio_id'];
+        $email = $credentials['email'];
+        $timestamp = $credentials['timestamp'];
+        $secret = getenv('INTEGRATEIO_SSO_SECRET');
+
+        $hashedToken = sha1($integrateio_id . ':' . $email . ':' . $secret . ':' . $timestamp);
+        return $hashedToken;
+    }
+     /**
+     * Performs login for Integrate Io Hosted trial Users.
+     *
+     * @param array $credentials
+     * @param bool  $remember
+     *
+     * @return array
+     * @throws BadRequestException
+     * @throws UnauthorizedException
+     * @throws \Exception
+     */
+    protected function handleIntegrateLogin(array $credentials = [], $remember = false)
+    {
+        // Check all params are there:
+        $requiredParams = array('integrateio_id', 'email', 'sso_token', 'timestamp');
+
+        // Make sure that all params are in the request.
+        foreach ($requiredParams as $requiredParam) {
+            if (!isset($credentials[$requiredParam])) {
+                throw new BadRequestException('Missing Parameters');
+            }
+        }
+
+        if ($this->generateToken($credentials) === $credentials['sso_token'] && ($credentials['timestamp'] > (time() - 120))) {
```

Comment:
@tomonorman Just a nitpick, let's wrap this part with parenthesis so that it's consistent with the latter condition
`($this->generateToken($credentials) === $credentials['sso_token'])`